File: src/IsbInterop.CSharp/Searches/Wrappers/SearchForObjectDescription.cs
using IsbInterop.Base;
using System;

namespace IsbInterop.Searches.Wrappers
{
  /// <summary>
  /// Обертка над ISearchForComObjectDescription.
  /// </summary>
  internal class SearchForObjectDescription : SearchDescription, ISearchForObjectDescription
  {
    /// <summary>
    /// Инициализирует поиск.
    /// </summary>
    /// <param name="objectInfo">Информация об объекте поиска.</param>
    public void InitializeSearch(IObjectInfo objectInfo)
    {
      if (objectInfo == null)
        throw new ArgumentNullException(nameof(objectInfo));

      InvokeRcwInstanceMethod("InitializeSearch", ((IRcwProxy)objectInfo).RcwObject);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwISearchForObjectDescription">COM-объект ISearchForObjectDescription.</param>
    /// <param name="scope">Область видимости.</param>
    internal SearchForObjectDescription(object rcwISearchForObjectDescription, IScope scope)
      : base(rcwISearchForObjectDescription, scope) { }
  }
}

File: src/IsbInterop/Searches/Proxies/SearchForObjectDescription.vb
Imports IsbInterop.Base

Namespace Searches.Proxies

    ''' <summary>
    ''' Обертка над ISearchForObjectDescription.
    ''' </summary>
    Friend Class SearchForObjectDescription
        Inherits SearchDescription
        Implements ISearchForObjectDescription

        ''' <summary>
        ''' Инициализирует поиск.
        ''' </summary>
        ''' <param name="objectInfo"></param>
        Public Sub InitializeSearch(objectInfo As IObjectInfo) Implements ISearchForObjectDescription.InitializeSearch
            If objectInfo Is Nothing Then
                Throw New ArgumentNullException("objectInfo")
            End If

            InvokeRcwInstanceMethod("InitializeSearch", DirectCast(objectInfo, IRcwProxy).RcwObject)
        End Sub

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="rcwISearchForObjectDescription">COM-объект ISearchForObjectDescription.</param>
        ''' <param name="scope">Область видимости.</param>
        Friend Sub New(rcwISearchForObjectDescription As Object, scope As IScope)
            MyBase.New(rcwISearchForObjectDescription, scope)
        End Sub
    End Class
End Namespace